elasticsearch_dsl/utils.py

```diff
@@ -468,7 +470,15 @@ def _clone(self) -> Self:
         return c
 
 
-class HitMeta(AttrDict[Any]):
+if TYPE_CHECKING:
+    HitMetaBase = HitBaseType
+else:
+    HitMetaBase = AttrDict[Any]
+
+
+class HitMeta(HitMetaBase):
+    inner_hits: Mapping[str, Any]
```

Comment:
I haven't found a way to type inner hints appropriately without making breaking changes. I think if we have the intetion to have proper typing we'll have to evaluate a change here.

For example, consider:

```python
hit.inner_hits["passages"][0].content
```

Here it is impossible to type the `content` attribute of the first `Passage` document, since `inner_hits` is a dictionary, and each entry in this dict maps to a different document type.

We may consider changing to something like this:

```python
hit.passages.meta.inner_hits[0].content
```

I think this may be possible to type because the inner hits belong to the document instead of being in a dict. But I'm not sure I want to get into such a big change now.